ttlcache/pointer_cache.go

```diff
@@ -0,0 +1,48 @@
+package ttlcache
+
+import (
+	"context"
+	"time"
+)
+
+// PointerCache holds pointers in a cache with a time-to-live.
+type PointerCache[T any] interface {
+	// Get gets the value with id with at least the minimum lifetime remaining.
+	Get(ctx context.Context, id string, minimumLifetime time.Duration) (*T, bool)
+	// Put adds a value to the cache with the given expiration time.
+	Put(ctx context.Context, id string, value *T, expiresAt time.Time)
+	// Delete removes the value with id from the cache. This is typically used
+	// to clean up expired values. It will no-op if the id is not found.
+	Delete(ctx context.Context, id string)
+}
+
+// convertPointerCacheToCache creates a wrapper around a PointerCache that
+// adapts it to the Cache interface.
+// This isn't intended to be used in production code as it will
+// allocate a new value on every Get call. It is only intended to be used
+// for testing purposes.
+// In production code, use the PointerCache directly.
+func convertPointerCacheToCache[T any](ptrCache PointerCache[T]) Cache[T] {
```

Comment:
Can we move this into the test file directly? That's an ideal way to indicate that a function is really just to make testing easier (the comment is good too though)

Reply:
Actually, did we implement this whole thing just so that we can call TestCache? Could we re-work TestCache to take in a boolean and then build the cacheFunc using a weak or not-weak cache based on that? We should avoid having so much code just for testing, especially since it means the tests themselves are weaker, for also testing this logic.

Reply:
I implemented this convert func just for testing, I moved it to a test file now. I don't think we can rework `testCache`  but I made a new `testPointerCache` that can be used for PointerCache's now instead of doing it in the `weak_test.go` file.

The PointerCache interface is to better/cleanly support the `WithPointerOtel` helper that takes in a PointerCache